### ISSUE TITLE
Input with `aria-controls` and `aria-expanded` leads to error `Element input is missing required attribute role`

### ISSUE THREAD

**Issue by codeflorist** [validator/validator#1803]
URL being validated or code to reproduce error:

```html
<input
  type="checkbox"
  aria-label="Toggle Full Navigation"
  aria-controls="full-navigation"
  aria-expanded="false"
/>
```

As far as i know, the used aria-attributes should be valid for an input. So i'm confused, why this is even throwing an error.

Apart from that, the error `Element input is missing required attribute role` does not make sense for me, since afaik the input should not require a `role`, but https://github.com/validator/validator/issues/498 states, that this is a general problem with the validator.

**Comment by sideshowbarker** [comment]
The error here is actually because the `aria-checked` attribute is missing. The ARIA spec requires it or checkboxes; see https://w3c.github.io/aria/#checkbox.

But as has been acknowledged in other issues, the error message which gets emitted in some cases like this one is not very helpful in correctly pinpointing what fix needs to be made.

But because that’s a general problem, and not specific to this particular issue, I’m going ahead and closing this — because other than the acknowledged deficiency in the error-message text, the checker is behaving as expected here.

**Comment by codeflorist** [comment]
> The error here is actually because the `aria-checked` attribute is missing. The ARIA spec requires it or checkboxes; see https://w3c.github.io/aria/#checkbox.

@sideshowbarker

A native `<input type="checkbox">` does not need an `aria-checked` attribute (and neither a `role` attribute), since it is the native HTML element for a checkbox.

The `aria-checked` attribute (and `role="checkbox"`) would be required for other elements behaving like a checkbox (e.g. `<span role='checkbox' aria-checked='true'>`).

**Comment by sideshowbarker** [comment]
OK, thanks for following up and yeah I can see that makes sense.

So I think the actual problem here was that the checker wasn’t allowing the `aria-expanded` attribute for implicit checkboxes.

So in https://github.com/validator/validator/commit/9594c9a5507131d983add836b70d66cc952e1a6a I fixed that — and I think that will fix the issue reported. If not, lemme know.
